### DISCUSSION TITLE
Should "script supporting elements" read "script and template"?

### DISCUSSION THREAD

**Discussion by estelle** [mdn/mdn-community#467] [Browser compatibility data]
Most non-empty elements support script-supporting elements (script and template), so we should either include it on all the elements that do (body, ol, li, hgroup, h1-h6, etc), or create a page for script supporting elements similar to global attributes (which are global to all element, not just "not empty" elements), and then refer to it as we refer to the page that explains script and template being able to be nested almost anywhere.

According to the spec, the following are explicitly listed as including script supporting elements in their content model:

* hgroup
* ol
* ul
* menu
* dl
* div
* picture
* table
* tbody
* thead
* tfoot
* tr
* select
* datalist
* optgroup

(this does not include elements that include them indirectly in "phrasing content").

This isn't a large list.
We can include this just on these elements (Will's preference)
Alternatively, since they are supported everywhere, we could make it global, with the empty element caveat (Estelle's preference)

(Converstation originally between @estelle and @wbamberg in https://github.com/mdn/content/issues/29827)

**Comment by Rumyra** [comment]
Thanks @estelle - is the in-page suggestion to add it to the 'Technical Summary' table as per the linked pr?

**Reply by estelle** [reply]
The suggestion is to either:

1. create a page on MDN entitled script supporting elements, and explain "template and script", so we don't link offsite from technical summary table for one sentence in the spec that says that, or
2. any time "script supporting element" is linked (or should be linked, but it's missing) from technical summary, replace "script supporting element" with "`<template>` and `<script>`" instead of linking to the spec of the page we did or did not create for (1).

**Reply by teoli2003** [reply]
Given that there are only 2 HTML elements and that it is unlikely that a 3rd one will be added in the next five years, I think we should go with suggestion 2 because:
- it is technically correct
- it is simpler for the readers as they don't need to understand the additional "script-supporting element" concept.

**Reply by Rumyra** [reply]
I agree and prefer option 2
